arrow-schema/src/datatype.rs

```diff
@@ -498,6 +498,27 @@ impl DataType {
                 }
             }
     }
+
+    /// Check to see if `self` is a superset of `other`
+    ///
+    /// If DataType is a nested type, then it will check to see if the nested type is a superset of the other nested type
+    /// else it will check to see if the DataType is equal to the other DataType
+    pub fn contains(&self, other: &Field) -> bool {
```

Comment:
```suggestion
    pub fn contains(&self, other: &DataType) -> bool {
```
I would expect this to take a `DataType`

Reply:
If we take a `DataType`, how could we call `Field::contains` following?

Reply:
```
match (self, other) {
    (DataType::Struct(f1), DataType::Struct(f2)) => f1.contains(f2),
    ....
}

```